BrushedMotorControllerFirmware/include/driver.c

```diff
@@ -109,33 +112,54 @@ void enter_boot(void)
 #pragma region TimerWorkCallback // timer interrupt internal functions
 static void update_speed_and_position_continuous(struct k_work *work)
 {
+	uint64_t diff_fwd = 0;
+	uint64_t diff_bck = 0;
+	uint64_t diff = 0;
+
 	for (enum ChannelNumber chnl = 0; chnl < CONFIG_SUPPORTED_CHANNEL_NUMBER; ++chnl) {
 		count_timer += 1; // debug only
 
-		uint64_t diff = 0;
-
 		// count encoder interrupts between timer interrupts:
-		if (drv_chnls[chnl].count_cycles > drv_chnls[chnl].old_count_cycles) {
-			diff = drv_chnls[chnl].count_cycles - drv_chnls[chnl].old_count_cycles;
+		if (drv_chnls[chnl].count_cycles_fwd > drv_chnls[chnl].old_count_cycles_fwd) {
+			diff_fwd = drv_chnls[chnl].count_cycles_fwd -
+				   drv_chnls[chnl].old_count_cycles_fwd;
+		} else {
+			diff_fwd = 0;
 		}
-		drv_chnls[chnl].old_count_cycles = drv_chnls[chnl].count_cycles;
-		enum MotorDirection dir;
+
+		if (drv_chnls[chnl].count_cycles_bck > drv_chnls[chnl].old_count_cycles_bck) {
+			diff_bck = drv_chnls[chnl].count_cycles_bck -
+				   drv_chnls[chnl].old_count_cycles_bck;
+		} else {
+			diff_bck = 0;
+		}
```

Comment:
would be nice to just count the cycles and make a diff as negative numbers as well. Is there something blocking us from doing that?